generate.mjs

```diff
@@ -0,0 +1,68 @@
+import {spawnSync} from 'node:child_process';
+import fs from 'node:fs';
+
+export function execute(cmd, args, options) {
+  const { error, status } = spawnSync(cmd, args, {...options || {}, shell: true, stdio: 'inherit'});
+  if (error) {
+    throw new(Error);
+  }
+  if (status !== 0) {
+    throw new Error(`${cmd} exited with status code: ${status}`);
+  }
+}
+
+/**
+ * Transform generated to something closer to what we want
+ * so we have less manual work to do.
+ */
+function fixupGenerated(filename) {
```

Comment:
Most (but not all) of these things would ideally be changed in the upstream generator. That repo is external but it was created for webgpu so we can change it freely.

Since some of these do make sense to keep here, it's not a big deal, since we need this script anyway. I'm fine with accepting it as is.